include/app-framework/Buffers/DequeBuffer.hh

```diff
@@ -38,6 +48,75 @@ namespace appframework{
     size_t fPopRetries;
   };
 
+
+template <class T>
```

Comment:
This is exactly why I asked asked about the extension of templated file we want to use, when we had this discussion  ages ago. Would it be possible to put these implementation in a file called DequeuBuffer.icc  that is included at the end of the header?

Reply:
If people agree that this a good idea, I personally have no problem with it. From my perspective, the positive is that you don't have tons of code right after the class declaration that a class's users (not developers) don't need to worry about, the negative is that developers would need to jump back and forth between the two files.